PolyEngine/Core/Src/Math/AABox.hpp

```diff
@@ -12,54 +12,92 @@ namespace Poly {
 	{
 	public:
 
-		/// <summary>Constructor from position and size.</summary>
-		/// <param name="position">Position of the min point of the box.</param>
+		/// <summary>Constructor from min and size.</summary>
+		/// <param name="min">Position of the min point of the box.</param>
 		/// <param name="size">Size (extent) of the box in each of the directions.</param>
-		AABox(const Vector& position, const Vector& size);
+		AABox(const Vector& min, const Vector& size);
 		AABox() = default;
 
 		/// <summary>Calculates center of the box.</summary>
 		/// <returns>Center of the box.</returns>
-		Vector GetCenter() const { return Pos + (Size * 0.5f); }
+		Vector GetCenter() const { return Min + (Size * 0.5f); }
 		
 		/// <summary>Returns the min point of the box. (posiiton)</summary>
 		/// <returns>Min point of the box.</returns>
-		const Vector& GetMin() const { return Pos; }
+		const Vector& GetMin() const { return Min; }
 
 		/// <summary>Returns the max point of the box.</summary>
 		/// <returns>Max point of the box.</returns>
-		Vector GetMax() const { return Pos + Size; }
+		Vector GetMax() const { return Min + Size; }
 		
 		/// <summary>Returns Size of the box in all three dimensions.</summary>
 		/// <returns>Size of the box. (extent)</returns>
 		const Vector& GetSize() const { return Size; }
 
 		/// <summary>Sets position of the box.</summary>
 		/// <param name="pos">New position</param>
-		void SetPosition(const Vector& pos) { Pos = pos; }
+		void SetMin(const Vector& pos) { Min = pos; }
 
 		/// <summary>Sets size of the box.</summary>
 		/// <param name="size">New size</param>
 		void SetSize(const Vector& size) { Size = size; }
 
+		/// <summary>Checks whether a given AABox is interecting with this AABox.</summary>
+		/// <param name="rhs">Other box.</param>
+		/// <see cref="AABox.Contains()"/>
+		inline bool Intersects(const AABox& rhs) const
+		{	
+			return (abs(Min.X - rhs.Min.X) * 2.0f < (Size.X + rhs.Size.X))
```

Comment:
There is `Vector::abs` method (or at least should be), use it.

Reply:
BasicMath.hpp has Poly::abs for float, double and long double overloads but no Vector.

Reply:
Also really needs abs or check on every axis is independent?

Reply:
Ok, I added this to trello.